### PR TITLE
Added page about SQL prepared statements

### PR DESCRIPTION
**Pull request by Zhaars**

### CODE REVIEW

File: content/acra/security-controls/sql-prepared-statements/_index.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+AcraServer fully supports syntax of prepared statements for PostgreSQL.
+
+For Prepare query like:
+
+```
+PREPARE fooplan (int, text, bool, numeric) AS INSERT INTO foo VALUES($1, $2, $3, $4);
+```
+
+AcraServer will detect the prepared statement query and store it destructed for further processing.
```
Comment (vixentael) [inline content/acra/security-controls/sql-prepared-statements/_index.md:29]: store it destructed?

**Review by vixentael — APPROVED** [review]

File: content/acra/security-controls/sql-prepared-statements/_index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+For example with query
+
+```
+SET @users__name = 'value'
+```
+
+and encryptor config:
+
+```
+schemas:
```
Comment (Lagovas) [inline content/acra/security-controls/sql-prepared-statements/_index.md:59]: let's change the order. at first show the encryptor config, after that show SET command. It is a bit easier to read and match in the mind, and logically correct. At first user should describe encryptor_config, then use this configuration in the own queries.

File: content/acra/security-controls/sql-prepared-statements/_index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+To overwrite the `delimiter` used in `SET` queries, `prepared_statements_set_arg_delimiter`
+setting could be used (if no setting provided the `__` delimiter will be used by default)
+
+```
+database_settings:
+  mysql:
+    prepared_statements_set_arg_delimiter: '&&'
+
+schemas:
+ .....
```
Comment (Lagovas) [inline content/acra/security-controls/sql-prepared-statements/_index.md:79]: let's place here the whole example schema. it hasn't much content to truncate it, but will be ready to copy/paste into own config file.

File: content/acra/security-controls/sql-prepared-statements/_index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+It is also possible to use [searchable encryption](/acra/security-controls/searchable-encryption/) with SQL prepared
+statements.
+AcraServer will analyze and transform queries as the following:
+
+```
+PREPARE stmt1 FROM 'SELECT * FROM test_table WHERE searchable_field = ?';
+```
+
+```
+PREPARE stmt1 FROM 'SELECT * FROM test_table WHERE substr(searchable_field, <search HASH size>, 1) = substr(?, <search HASH size>, 1)';
```
Comment (Lagovas) [inline content/acra/security-controls/sql-prepared-statements/_index.md:91]:
```suggestion
PREPARE stmt1 FROM 'SELECT * FROM test_table WHERE substr(searchable_field, 1, <search HASH size>) = substr(?, 1, <search HASH size>)';
```

Comment (Lagovas) [reply content/acra/security-controls/sql-prepared-statements/_index.md:91]: looks like here we pass arguments into the substr in wrong order

**Review by Lagovas — CHANGES_REQUESTED** [review]

File: content/acra/security-controls/sql-prepared-statements/_index.md (hunk 1 above)
Comment (vixentael) [reply content/acra/security-controls/sql-prepared-statements/_index.md:29]: +1 to @Lagovas

**Review by Lagovas — APPROVED** [review]

**Comment by vixentael** [comment]
let's meeeerge and push to prod?

**Pull request merged**